### PR TITLE
Add tests for input and change events on checkboxes and radios

### PR DESCRIPTION
**Pull request by josepharhar**
This follows @domenic's suggestion in https://github.com/whatwg/html/issues/5291

### CODE REVIEW

File: dom/events/Event-dispatch-detached-input-and-change.html [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+<meta charset="utf-8">
+<link rel="author" title="Joey Arhar" href="mailto:jarhar@chromium.org">
+<title>input and change events for detached checkbox and radio elements</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+
+<body>
+<script>
+
+async_test(t => {
```
Comment (domenic) [inline dom/events/Event-dispatch-detached-input-and-change.html]: It looks like these can all be normal sync `test()`s, with no need for `t` or `t.done()`.

File: dom/events/Event-dispatch-detached-input-and-change.html [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+
+  let inputEventFired = false;
+  input.addEventListener('input', () => inputEventFired = true);
+  let changeEventFired = false;
+  input.addEventListener('change', () => changeEventFired = true);
+  input.click();
+  assert_true(inputEventFired);
+  assert_true(changeEventFired);
+
+  input.remove();
```
Comment (domenic) [inline dom/events/Event-dispatch-detached-input-and-change.html]: To make cleanup happen even if the test fails, you should move this up to line 76 and wrap it in `t.add_cleanup(() => input.remove())`.

However, it's also fairly unnecessary. A few extra elements in the DOM at the end of the test doesn't hurt anyone. So you could also just remove these.

File: dom/events/Event-dispatch-detached-input-and-change.html [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
```
Comment (domenic) [inline dom/events/Event-dispatch-detached-input-and-change.html]: For these tests, it's not really necessary to remove both the input and the shadow host; removing the shadow host will remove its children too.

**Review by domenic — APPROVED** [review]
This is awesome. Some minor test style nits, but the actual contents of the test and the exhaustiveness of them is top-notch.

File: dom/events/Event-dispatch-detached-input-and-change.html (hunk 2 above)
Comment (josepharhar) [reply dom/events/Event-dispatch-detached-input-and-change.html]: I removed the cleanup

**Comment by josepharhar** [comment]
I can't merge this because I don't have write access to this repository.
@domenic can you merge this for me?

**Pull request merged**